app/_hub/kong-inc/openid-connect/how-to/client-authentication/_mtls.md

```diff
@@ -0,0 +1,108 @@
+---
+title:  Mutual TLS Client Authentication
+nav_title: Mutual TLS Client Authentication
+---
+
+## Mutual TLS client authentication
+
+The OpenID Connect plugin supports mutual TLS (mTLS) client authentication with the IdP. 
+When mTLS authentication is enabled, Kong establishes mTLS connections with the IdP using the configured client certificate.
+
+You can use mTLS client with the following IdP endpoints and corresponding flows:
+
+* `token`
+  * [Authorization Code Flow](/hub/kong-inc/openid-connect/how-to/authentication/authorization-code-flow/)
+  * [Password Grant](/hub/kong-inc/openid-connect/how-to/authentication/password-grant/)
+  * [Refresh Token Grant](/hub/kong-inc/openid-connect/how-to/authentication/refresh-token/)
+* `introspection`
+  * [Introspection Authentication flow](/hub/kong-inc/openid-connect/how-to/authentication/introspection/)
+* `revocation`
+  * [Session Authentication](/hub/kong-inc/openid-connect/how-to/authentication/session/)
+
+For all these endpoints and for the flows supported, the plugin uses mTLS client authentication as the authentication method when communicating with the IdP, for example, to fetch the token from the token endpoint.
+
+## Requirements for an mTLS client auth configuration
+
+### Auth server configuration
+
+The feature requires the IdP to be enabled to use mTLS and X.509 client certificate authentication. For configuring this in Keycloak, refer to the [Keycloak configuration](#prerequisites) section below. For different auth servers, refer to their documentation to configure this functionality.
+
+### Kong configuration
+
+* Set the client authentication methods. Refer to the configuration reference sections:
+	* [`client_auth`](/hub/kong-inc/openid-connect/configuration/#configclient_auth)
+	* [`token_endpoint_auth_method`](/hub/kong-inc/openid-connect/configuration/#configtoken_endpoint_auth_method)
+	* [`introspection_endpoint_auth_method`](/hub/kong-inc/openid-connect/configuration/#configintrospection_endpoint_auth_method)
+	* [`revocation_endpoint_auth_method`](/hub/kong-inc/openid-connect/configuration/#configrevocation_endpoint_auth_method)
+* Configure the client certificate ID as a value for the `tls_client_auth_cert_id` field as described in the [configuration reference](/hub/kong-inc/openid-connect/configuration/#config-tls_client_auth_cert_id).
+
+#### Certificates
+
+For mTLS connections, you need to pass a certificate and key to {{site.base_gateway}}. 
```

Comment:
Not exactly, rather, for the gateway to be able to establish mTLS connections with the IdP it is necessary to configure a client certificate and key, to be used. Once the feature is configured the client does not pass certificates to the gateway, it is a one-time configuration so that the gateway then has what it needs to establish mTLS connections with the IdP later on.

Basically this step is instructing how to create a new certificate and key pair in the gateway to be used by this feature when connecting with the IdP.